packages/grafana-ui/src/components/FilterPill/FilterPill.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useContext } from 'react';
+import { stylesFactory, ThemeContext } from '../../themes';
+import { GrafanaTheme } from '@grafana/data';
+import { css } from 'emotion';
+import { IconButton } from '../IconButton/IconButton';
+import { IconName } from '../../types';
+
+interface FilterPillProps {
+  selected: boolean;
+  label: string;
+  onClick: React.MouseEventHandler<HTMLElement>;
+  icon?: IconName;
+}
+
+export const FilterPill: React.FC<FilterPillProps> = ({ label, selected, onClick, icon = 'check' }) => {
+  const theme = useContext(ThemeContext);
+  const styles = getFilterPillStyles(theme, selected);
+  return (
+    <div className={styles.wrapper} onClick={onClick}>
+      <IconButton
+        name={icon}
+        onClick={e => {
+          e.stopPropagation();
+          onClick(e);
+        }}
+        className={styles.icon}
+        surface="header"
+      />
+      <span className={styles.label}>{label}</span>
+    </div>
+  );
+};
+
+const getFilterPillStyles = stylesFactory((theme: GrafanaTheme, isSelected: boolean) => {
+  const labelColor = isSelected ? theme.colors.text : theme.colors.textWeak;
+
+  return {
+    wrapper: css`
+      padding: ${theme.spacing.xxs} ${theme.spacing.sm};
```

Comment:
This class look like it duplicates the gf-form-label styles. If so maybe we should use that instead until we have a InlineFormLabel component we can reuse

Reply:
This will never be an inline form label IMO as this is not a label. Gonna stick with this duplication here :)